packages/ui/src/elements/ReactSelect/DropdownIndicator/index.tsx

```diff
@@ -0,0 +1,34 @@
+import type { DropdownIndicatorProps } from 'react-select'
+
+import { Chevron } from '@payloadcms/ui/icons/Chevron'
+import React from 'react'
+
+import type { Option as OptionType } from '../types.js'
+
+import './index.scss'
+
+const baseClass = 'dropdown-indicator'
+
+export const DropdownIndicator: React.FC<DropdownIndicatorProps<OptionType, true>> = (props) => {
+  const {
+    innerProps: { ref, ...restInnerProps },
+  } = props
+
+  return (
+    <div
+      className={baseClass}
+      ref={ref}
+      {...restInnerProps}
+      onKeyDown={(e) => {
+        if (e.key === 'Enter') {
+          // trigger the menu to open
+          return (e.key = ' ')
+        }
```

Comment:
I think you could just reassign the event to space in this case.

```ts
// allow for enter to also allow opening of the modal
if (e.key === 'Enter') e.key = ' '
```

in this case you should run theirs before yours.